accepted/2020/net5/net5.md

```diff
@@ -562,14 +562,26 @@ Specifically:
 * We'll follow the rule-based creation in SDK-style projects which takes the
   friendly TFM name without the OS flavor, makes it upper-case and replaces
   special characters with an underscore.
-* We'll only define versionless symbols for the OS flavor (because
-  multi-targeting between OS versions will be rare).
-
-_**Open Issue**. Review defines with Xamarin folks._
-
-_**Open Issue**. Should we also define symbols for previous versions? For
-example, on .NET 6, should we define `NET6_0` as well as `NET6_0_OR_HIGHER`, and
-`NET5_0_OR_HIGHER`? This could simplify `#if` logic._
+* For OS flavors we'll do the same, i.e. create versionless as well as version
+  specific symbols.
+
+In order to make it easier to update code, especially when doing
+multi-targeting, we should make them additive, so that when targeting `net6.0`
+both `NET6_0` and `NET5_0` are defined. The same applies to OS bindings.
+Examples:
+
+* `net5.0`
+  * `NETCOREAPP`, `NETCOREAPP3_1` (for backwards compatibility)
+  * `NET`, `NET5_0`  
+* `net6.0`
+  * (same as `net5.0`)
+  * `NET6_0`  
```

Comment:
Have in mind that there may be *a lot* of these defines, since there will be one for every release.

Say you have this code:

```csharp
#if IOS14_0
// Do something awesome Apple added support for in iOS 14
#endif
```

In your approach that code would stop working with iOS 14.1, since then we'd define `IOS14_1`, and not `IOS14_0`.

If you want something in a specific version only, you could do:

```csharp
#if NET_5_0 && !NET_6_0
// Code specific to net5.0...
#endif
```
```

Reply:
> Have in mind that there may be a lot of these defines, since there will be one for every release.

It'll be there even if I don't target `net5`, `net6` when I'm targeting `net7`. This is why I don't want this. The defines should be specific to the binding packages and it's version.

> In your approach that code would stop working with iOS 14.1, since then we'd define `IOS14_1`, and not `IOS14_0`.

You could define `IOS14` in addition to those. But that's not the point. The defines, including platform and version specific props should come from the bindings package.

__Putting everything in the SDK is not an answer. It just leads to more bloat.__

Reply:
As I said above, the problem with the current approach is that it breaks code as soon as the version is added (in multi-targeting) or bumped. We don't want this. Yes, this means that the SDK will have to grow but I don't think it's that bad because if defined correct, they can be chained.